### PR TITLE
Survey Template Rework

### PR DESCRIPTION
**Pull request by CollinWoo**
### Summary 

This pull request implements the reworked survey questions and step template design shown [here](https://www.figma.com/file/9zueLCvKT7BBMYi8Dt8Ukp/Zing-LSC?node-id=2394%3A234446). See the second screen onwards from http://localhost:3000/survey.

- [x] updated survey questions in Questions.json by rewording response options and deleting COVID-specific questions
- [x] updated buttons and progress bar to match the Figma design
- [x] made it so scrollbars are added if survey options overflow the card (this wasn't specified in the design but I thought it made sense) 
- [x] attempted to keep the design responsive, although I'm unsure about how well I accomplished this 

### Test Plan 

Changes were mostly visual so testing was done by comparing the Figma to the actual frontend. Screenshots of the frontend implementation are below:

Screen Shot 2022-04-17 at 1 53 06 PM
Screen Shot 2022-04-17 at 1 54 00 PM

### Notes 
- most of the design changes were accomplished with inline styling in StepTemplate.tsx. This is probably not good practice but I was too afraid to touch the existing styling for components.
- Some of the non-template related styling (text boxes, radio buttons) does not match the Figma. I only edited the survey template so these were not updated.

### Issues
- Removing survey questions from Questions.json seems to have broken some parts of the backend, since some fields are now undefined

### CODE REVIEW

File: frontend/src/modules/Core/Questions/Questions.json [hunk 1]
```diff
@@ -3,53 +3,24 @@
...
-      "ILR": "Industrial and Labor Relations",
-      "SC Johnson": "SC Johnson College of Business",
-      "Engineering": "Engineering"
+      "CALS": "CALS",
+      "AAP": "AAP",
+      "A&S": "CAS",
+      "Hotel Admin": "Hotel Administration",
+      "Dyson": "Dyson School",
+      "Engineering": "Engineering",
+      "ILR": "ILR"
```
Comment (Coreball) [inline frontend/src/modules/Core/Questions/Questions.json]: HumEc should not have been deleted. If you notice something unusual about the designs make sure to ask! I talked to Alyssa and it should have been there. A&S > CAS officially, let's prefer that here too.

File: frontend/src/modules/Survey/Components/StepTemplate.tsx [hunk 2]
```diff
@@ -37,25 +32,37 @@ export const StepTemplate: FunctionComponent<StepTemplateProps> = ({
...
+          style={{ height: '15%', color: '#815ed4', margin: '0% 2%' }}
+        >
+          <div style={{ textAlign: 'center' }}>
+            <IconButton className="prev" onClick={handlePrev} color="secondary">
+              <ArrowBack />
+            </IconButton>
+            <div>Prev</div>
+          </div>
+
+          <div style={{ marginLeft: 'auto', textAlign: 'center' }}>
```
Comment (Coreball) [inline frontend/src/modules/Survey/Components/StepTemplate.tsx]: I'm not sure how I feel about raw divs being used and styled this way. It definitely works, but I am hesitant towards adding a third styling solution on top of the two we're already using, Styled and MUI `sx`. Ideally, we would have one styling solution used everywhere to make things simpler (or a few, if the situations in which each is preferred are clearly delineated).

**Review by Coreball — CHANGES_REQUESTED** [review]
Good work updating the survey questions and figuring out how to use MUI components such as IconButton! Thanks for the screenshots and detailed notes in the PR, I have a few comments on them in addition to the comments I made in the code.

For the design changes, try to be un-afraid to touch the existing styling/components! The faster you do this, the quicker you'll learn how to use them + get familiar coding in the project.

We should also avoid having PRs that would break main in a significant way when merged (or have the PR that would fix things ready to go). This PR should include the changes to handle the reduced question set on the backend, provided they are not too major. When people branch off of main later, it would be best if they are able to work on new code and test it without running into problems submitting the survey. This is especially important for this PR because student survey submissions form the basis for most other testing.

File: frontend/src/modules/Core/Questions/Questions.json (hunk 1 above)
Comment (CollinWoo) [reply frontend/src/modules/Core/Questions/Questions.json]: kk fixed the questions to include humec and changed CAS to A&S

**Comment by Coreball** [comment]
Also forgot to add this but the scroll looks kind of weird on my computer (macOS, Firefox)

Screen Shot 2022-04-19 at 22 23 27

File: backend/functions/src/matching/functions.ts [hunk 3]
```diff
@@ -85,9 +85,10 @@ async function makeMatches(courseId: string, groupSize = 3) {
         })
     )
   )
-  studentData.sort(
-    (obj1, obj2) => obj1.preferredWorkingTime - obj2.preferredWorkingTime
-  )
+  studentData
+    .sort
+    //(obj1, obj2) => obj1.preferredWorkingTime - obj2.preferredWorkingTime
+    ()
```
Comment (Coreball) [inline backend/functions/src/matching/functions.ts]: You can remove the sort entirely

**Review by Coreball — APPROVED** [review]
Good work, let's just get this merged in to keep this PR small and focused. We can clean up other parts of the survey later. Since there are conflicts, you'll have to merge main into your branch to fix them.

File: frontend/src/modules/Survey/Components/StepBegin.tsx [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import React, { useState } from 'react'
+import { IconButton, Box } from '@mui/material'
```
Comment (jessica0801) [inline frontend/src/modules/Survey/Components/StepBegin.tsx:2]: Nice use of the mui icons! it makes the code much simpler and the  look more consistent

**Review by jessica0801 — COMMENTED** [review]
great job on this PR!! updating all the fields and information was incredibly important, and I am excited to see this product be updated and ready for use. I also loved the   incorporation of MUI into a lot of these elements.

**Pull request merged**